tests/conftest.py

```diff
@@ -13,34 +13,82 @@
 #  permissions and limitations under the License.
 import logging
 import os
+import shutil
+import sys
 
 import pytest
 
 from zenml.artifacts.base_artifact import BaseArtifact
 from zenml.constants import ENV_ZENML_DEBUG
 from zenml.core.repo import Repository
-from zenml.exceptions import InitializationException
 from zenml.materializers.base_materializer import BaseMaterializer
 from zenml.pipelines import pipeline
 from zenml.steps import StepContext, step
 
 
-def pytest_sessionstart(session):
-    """Called after the Session object has been created and
-    before performing collection and entering the run test loop.
-    """
+@pytest.fixture(scope="session", autouse=True)
+def base_repo(tmp_path_factory, session_mocker):
+    """Fixture to get a base clean repository for all tests."""
+    # original working directory
+    orig_cwd = os.getcwd()
+
+    # set env variables
     os.environ[ENV_ZENML_DEBUG] = "true"
-    try:
-        Repository.init_repo()
-    except InitializationException:
-        # already initialized
-        logging.info("Repo already initialized for testing.")
+    os.environ["ZENML_ANALYTICS_OPT_IN"] = "false"
+
+    # change the working directory to a fresh temp path
+    tmp_path = tmp_path_factory.mktemp("tmp")
+    os.chdir(tmp_path)
+
+    # patch the global dir just within the scope of this function
+    logging.info(f"Tests are running in repo path: {tmp_path}")
+    session_mocker.patch.object(
+        sys.modules["zenml.io.utils"],
+        "get_global_config_directory",
+        return_value=str(tmp_path / "zenml"),
+    )
+
+    # initialize repo at path
+    Repository.init_repo(str(tmp_path))
+    repo = Repository(str(tmp_path))
+
+    # monkey patch original cwd in for later use and yield
+    repo.original_cwd = orig_cwd
+    yield repo
+
+    # clean up
+    shutil.rmtree(tmp_path)
+
+
+@pytest.fixture
+def clean_repo(tmp_path_factory, mocker, base_repo: Repository):
+    """Fixture to get a clean repository for an individual test."""
+    # set env variables
+    os.environ[ENV_ZENML_DEBUG] = "true"
```

Comment:
Can we get away with removing this duplicate code as this is calling the base fixture anyway?

Reply:
Indeed. Removed it in my latest commit! Please feel free to check and approve and ill merge :-)